dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt

```diff
@@ -560,10 +566,12 @@ internal class CoreFeature(
     }
 
     private fun setupExecutors() {
-        @Suppress("UnsafeThirdPartyFunctionCall") // pool size can't be <= 0
-        uploadExecutorService = LoggingScheduledThreadPoolExecutor(CORE_DEFAULT_POOL_SIZE, internalLogger)
-        @Suppress("UnsafeThirdPartyFunctionCall") // workQueue can't be null
-        persistenceExecutorService = persistenceExecutorServiceFactory(internalLogger)
+        uploadExecutorService = LoggingScheduledThreadPoolExecutor(
```

Comment:
should we use `scheduledExecutorServiceFactory` for this one?

Reply:
Unfortunately because we rely on it being a specific implementation to drain it (when we call the flush method), we can't use the generic factory which returns a `ScheduledExecutorService`. And I don't want to polute the factory by coupling tightly to a specific class.

Reply:
In that case why do we use the `scheduledExecutorServiceFactory` ?

Reply:
Nevermind, I got my answer below ;)